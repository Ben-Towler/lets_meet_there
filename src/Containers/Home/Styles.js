import styled from 'styled-components'

export const Wrapper = styled.section`
`;

export const Hero = styled.section`
  background-color: orange;
  
`;

export const Title = styled.section`
`;