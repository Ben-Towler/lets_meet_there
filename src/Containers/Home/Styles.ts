import styled from 'styled-components'

export const Wrapper = styled.section`
  max-width: 80%;
  margin: 0 auto;
`;

export const TitleWrapper = styled.section`
  margin: 1em;
  padding: 1em;
`;

export const Hero = styled.section`
  height: 90vh;
`;

export const Title = styled.section`
  font-family: ${props => props.theme.fonts[0]};
  font-size: ${props => props.theme.fontSizes.large};
  color: ${props => props.theme.colors.navyBlue};
`;

export const SubTitle = styled.p`
  font-family: ${props => props.theme.fonts[0]};
  font-size: ${props => props.theme.fontSizes.small};
  color: ${props => props.theme.colors.gullGrey};
`;