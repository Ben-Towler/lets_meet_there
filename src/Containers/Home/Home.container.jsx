import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ApiClient from 'Services/ApiClient';
import { SearchForm, Spinner } from 'Components';
import { FlightList } from 'Containers';
import helpers from 'helpers';
import { isLoading, getPlaces, getCarriers } from 'Actions';
import { Wrapper, Hero, Title, TitleWrapper, SubTitle } from './Styles';

const style = {
  'text-decoration': 'underline',
  'text-decoration-color': '#5FDAE3'
}

export default function HomePage({currentUser}) {

  const [matched, setMatched] = useState([]);
  // Redux items
  const loading = useSelector((state) => state.isLoading);
  const dispatch = useDispatch();

  const getPlace = async (query) => {
    return ApiClient.getPlace(query).then((res) => {
      return helpers.placeId(res, query);
    });
  };

  const searchFlights = async (from1, from2, departDate, returnDate) => {
    dispatch(isLoading());
    const quotesA = await ApiClient.getFlights(from1, departDate, returnDate);
    const quotesB = await ApiClient.getFlights(from2, departDate, returnDate);
    dispatch(getPlaces(quotesA.places, quotesB.places)); // dispatch is a redux function that gets the named reducer and sets the state.
    dispatch(getCarriers(quotesA.carriers, quotesB.carriers));
    setMatched(helpers.matchFlights(quotesA, quotesB)); // as this is passed through props fine to leave outside redux.
    dispatch(isLoading());
  };

  return (
    <Wrapper>
      <Hero>
        <TitleWrapper>
          <SubTitle>PLAN THE PERFECT TRIP TOGETHER</SubTitle>
          <Title>Search for a place to <span style={style}>meet</span></Title>
        </TitleWrapper>
        <SearchForm
          searchFlights={searchFlights}
          getPlace={getPlace}
        />

        {<div>
          {!loading 
            ? <FlightList user={currentUser} matchedFlights={matched}></FlightList>
            : <Spinner />
          }
        </div>}
      </Hero>
    </Wrapper>
  );
};
