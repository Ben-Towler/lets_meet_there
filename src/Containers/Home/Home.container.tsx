import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getPlace as apiGetPlace, getFlights } from 'Services/ApiClient';
import { SearchForm, Spinner } from 'Components';
import { FlightList } from 'Containers';
import helpers from 'helpers';
import { isLoading, getPlaces, getCarriers } from 'Actions';
import { Wrapper, Hero, Title } from './Styles';

interface Props {
  currentUser?: any;
}

export default function HomePage({currentUser}: Props): JSX.Element {

  const [matched, setMatched] = useState([]);
  // Redux items
  const loading = useSelector((state) => state.isLoading);
  const dispatch = useDispatch();

  const getPlace = async (query) => {
    return apiGetPlace(query).then((res) => {
      return helpers.placeId(res, query);
    });
  };

  const searchFlights = async (from1, from2, departDate, returnDate) => {
    dispatch(isLoading());
    const quotesA = await getFlights(from1, departDate, returnDate);
    const quotesB = await getFlights(from2, departDate, returnDate);
    dispatch(getPlaces(quotesA.places, quotesB.places)); // dispatch is a redux function that gets the named reducer and sets the state.
    dispatch(getCarriers(quotesA.carriers, quotesB.carriers));
    setMatched(helpers.matchFlights(quotesA, quotesB)); // as this is passed through props fine to leave outside redux.
    dispatch(isLoading());
  };

  return (
    <Wrapper>
      <Hero>
        <Title>Search for a place to meet</Title>
        <SearchForm
          searchFlights={searchFlights}
          getPlace={getPlace}
        />

        {<div>
          {!loading 
            ? <FlightList user={currentUser} matchedFlights={matched}></FlightList>
            : <Spinner />
          }
        </div>}
      </Hero>
    </Wrapper>
  );
};
