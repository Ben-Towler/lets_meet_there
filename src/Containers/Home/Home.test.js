import React from 'react'
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from '../Home/Home.container';

const mockStore = configureStore([]);

const store = mockStore()

describe('test the Spinner', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Provider store={store}><Home /></Provider>);
    expect(asFragment(<Home />)).toMatchSnapshot();
  })

  it('should render correctly', () => {
    const { getByText } = render(<Provider store={store}><Home /></Provider>);
    const title = getByText('Search for a place to meet');
    expect(title).toBeInTheDocument();
  })

})