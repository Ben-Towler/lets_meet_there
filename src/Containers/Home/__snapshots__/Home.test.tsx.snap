// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Spinner should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-fzozJi gJiFqy"
  >
    <section
      class="sc-fzoLsD kGNGgW"
    >
      <section
        class="sc-fzpans jgULOI"
      >
        Search for a place to meet
      </section>
      <div
        class="sc-AxirZ dqVvws"
      >
        <form
          class="sc-AxhUy elBAgh"
          data-testid="form"
        >
          <div>
            <div
              class="autocomplete-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                autocomplete="off"
                class="form-control"
                name="location"
                placeholder="London..."
                role="combobox"
                title="Location"
                type="text"
                value=""
              />
              <div
                class="autocomplete-dropdown-container"
              />
            </div>
          </div>
          <div>
            <div
              class="autocomplete-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                autocomplete="off"
                class="form-control"
                name="destination"
                placeholder="Barcelona..."
                role="combobox"
                title="Destination"
                type="text"
                value=""
              />
              <div
                class="autocomplete-dropdown-container"
              />
            </div>
          </div>
          <div
            class="DateRangePicker DateRangePicker_1"
          >
            <div>
              <div
                class="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2 DateRangePickerInput__showClearDates DateRangePickerInput__showClearDates_3"
              >
                <div
                  class="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-startDate"
                    aria-label="Start Date"
                    autocomplete="off"
                    class="DateInput_input DateInput_input_1"
                    id="startDate"
                    name="startDate"
                    placeholder="Start Date"
                    type="text"
                    value=""
                  />
                  <p
                    class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-startDate"
                  >
                    Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <div
                  aria-hidden="true"
                  class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                  role="presentation"
                >
                  <svg
                    class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                    focusable="false"
                    viewBox="0 0 1000 1000"
                  >
                    <path
                      d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                    />
                  </svg>
                </div>
                <div
                  class="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-endDate"
                    aria-label="End Date"
                    autocomplete="off"
                    class="DateInput_input DateInput_input_1"
                    id="endDate"
                    name="endDate"
                    placeholder="End Date"
                    type="text"
                    value=""
                  />
                  <p
                    class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-endDate"
                  >
                    Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  aria-label="Clear Dates"
                  class="DateRangePickerInput_clearDates DateRangePickerInput_clearDates_1 DateRangePickerInput_clearDates_default DateRangePickerInput_clearDates_default_2 DateRangePickerInput_clearDates__hide DateRangePickerInput_clearDates__hide_3"
                  type="button"
                >
                  <svg
                    class="DateRangePickerInput_clearDates_svg DateRangePickerInput_clearDates_svg_1"
                    focusable="false"
                    viewBox="0 0 12 12"
                  >
                    <path
                      d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <button
            class="sc-AxhCb lnpmzE"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div>
        <div>
          <div
            class="tile_Container"
          />
        </div>
      </div>
    </section>
  </section>
</DocumentFragment>
`;
